### PR TITLE
Adding Trivy security scanning to GitHub Action for building Docker images

### PR DESCRIPTION
**Pull request by mfenner**
### Description

Running Trivy image scanning in the GitHub Action instead of downstream in the repo (Harbor). Also running Trivy scan of the repo, which detects misconfigurations in Dockerfiles. Security alerts are pushed to the GitHub Security tab for reporting. Pushing built images to CERN repo, if not configured (e.g. in pull requests), use GitHub Container Registry.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [x] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/best-practices/i18n/) (for relevant code).
- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines (for relevant code).
- [x] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines (for relevant code).
- [x] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines (for relevant code).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).

**Third-party code**

If you've added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

**Comment by mfenner** [comment]
Fixed 26 Trivy security alerts from Dockerfile misconfigurations

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -37,16 +79,18 @@ jobs:
...
         id: meta
         uses: docker/metadata-action@v4
         with:
-          images: registry.cern.ch/${{ github.repository_owner }}/almalinux
+          images: |
+            ${{ env.REGISTRY }}/${{ github.repository_owner }}/almalinux
           tags: |
-            type=semver,pattern={{major}}
-            type=semver,pattern={{major}}.{{minor}}
+            # type=semver,pattern={{major}}
```
Comment (ntarocco) [inline .github/workflows/release.yml]: Why commenting this out?

**Review by ntarocco — COMMENTED** [review]
Thanks a lot for this PR, and for the various improvements in the Docker files.
We already have Trivy scan images on CERN registry, which runs on each push and already notifies us.
Why adding it also here? The benefit is indeed to push it to the GitHub Security tab, but such panel is visible to only who has access to it.

**Comment by mfenner** [comment]
There are two reasons to add the Trivy scans here:
* the harbor Trivy configuration misses the Trivy configuration scan that detects Dockerfile issues (I found 26 and fixed them)
* this puts the security scan further upstream, e.g. can work on pull requests.

File: .github/workflows/release.yml (hunk 1 above)
Comment (mfenner) [reply .github/workflows/release.yml]: I am not sure whether we want/need a tag for major version.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -26,6 +28,46 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3
 
+      - name: Run Trivy static code analysis
```
Comment (avivace) [inline .github/workflows/release.yml]: What is the target of this step here? Is this scanning every dockerfile in the repository?
Is this the step that runs the "trivy configuration scan that detects Dockerfile issues" you mentioned before?

**Review by avivace — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -26,6 +28,46 @@ jobs:
...
+          category: 'fs'
+
+      - name: Build an image from Dockerfile
+        run: |
+          docker build -t ghcr.io/${{ github.repository_owner }}/almalinux:${{ github.sha }} ./almalinux/
+
+      - name: Run Trivy image scan
+        uses: aquasecurity/trivy-action@master
+        with:
+          image-ref: 'ghcr.io/${{ github.repository_owner }}/almalinux:${{ github.sha }}'
```
Comment (avivace) [inline .github/workflows/release.yml]: Shouldn't this one be configured to use the registry variable so we don't have to expect to use the github packages registry?

**Review by avivace — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml (hunk 2 above)
Comment (mfenner) [reply .github/workflows/release.yml]: Yes, looks at every Dockerfile, also the deprecated ones.

File: .github/workflows/release.yml (hunk 3 above)
Comment (mfenner) [reply .github/workflows/release.yml]: It uses the image built in this GitHub action (in the previous step).

**Comment by ntarocco** [comment]
Thank you @mfenner for your hard work here! Very much appreciated.
I would propose to merge this, given the changes, and optimize the build in a next PR (I hope @avivace agrees): the idea is to be able to build images on tag pushed but also to rebuild them on the latest committed tag (some work is required).

**Comment by ntarocco** [comment]
@mfenner can you please fix the conflict?

**Comment by ntarocco** [comment]
Or eventually, feel free to change this PR to implement https://github.com/inveniosoftware/docker-invenio/issues/68

**Pull request closed without merging**